lib/experimental/class-theme-json-webfonts-handler.php

```diff
@@ -0,0 +1,537 @@
+		 *
+		 * @since 6.0.0
+		 */
+		class Theme_JSON_Webfonts_Handler {
+
+			/**
+			 * Webfonts to be processed.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @var array[]
+			 */
+			private $webfonts = array();
+
+			/**
+			 * Stylesheet handle.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @var string
+			 */
+			private $stylesheet_handle = '';
+
+			/**
+			 * Initialize the handler.
+			 *
+			 * @since 6.0.0
+			 */
+			public function init() {
+				add_action( 'wp_loaded', array( $this, 'register_webfonts' ) );
+
+				// Register callback to generate and enqueue styles.
+				if ( did_action( 'wp_enqueue_scripts' ) ) {
+					$this->stylesheet_handle = 'wp-webfonts-footer';
+					$hook                    = 'wp_print_footer_scripts';
+				} else {
+					$this->stylesheet_handle = 'wp-webfonts';
+					$hook                    = 'wp_enqueue_scripts';
+				}
+				add_action( $hook, array( $this, 'generate_and_enqueue_styles' ) );
+				add_action( 'admin_init', array( $this, 'generate_and_enqueue_editor_styles' ) );
+
+				add_filter( 'mime_types', array( $this, 'add_mime_types' ) );
+			}
+
+			/**
+			 * Register webfonts within the `theme.json`.
+			 *
+			 * @since 6.0.0
+			 */
+			public function register_webfonts() {
+				$this->webfonts = array();
+				$webfonts       = $this->get_webfonts_from_theme_json();
+
+				foreach ( $webfonts as $webfont ) {
+					if ( ! is_array( $webfont ) ) {
+						continue;
+					}
+
+					$webfont = $this->convert_keys_to_kebab_case( $webfont );
+
+					$webfont = $this->validate_webfont( $webfont );
+
+					$webfont['src'] = $this->transform_src_into_uri( (array) $webfont['src'] );
+
+					// Skip if not valid.
+					if ( empty( $webfont ) ) {
+						continue;
+					}
+
+					$this->webfonts[] = $webfont;
+				}
+			}
+
+			/**
+			 * Gets the webfonts from `theme.json`.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @return array Array of defined webfonts.
+			 */
+			private function get_webfonts_from_theme_json() {
+				// Get settings from theme.json.
+				$theme_settings = WP_Theme_JSON_Resolver::get_merged_data()->get_settings();
+
+				// Bail out early if there are no settings for webfonts.
+				if ( empty( $theme_settings['typography'] ) || empty( $theme_settings['typography']['fontFamilies'] ) ) {
+					return array();
+				}
+
+				$webfonts = array();
+
+				// Look for fontFamilies.
+				foreach ( $theme_settings['typography']['fontFamilies'] as $font_families ) {
+					foreach ( $font_families as $font_family ) {
+
+						// Skip if fontFace is not defined.
+						if ( empty( $font_family['fontFace'] ) ) {
+							continue;
+						}
+
+						// Skip if fontFace is not an array of webfonts.
+						if ( ! is_array( $font_family['fontFace'] ) ) {
+							continue;
+						}
+
+						$webfonts = array_merge( $webfonts, $font_family['fontFace'] );
+					}
+				}
+
+				return $webfonts;
+			}
+
+			/**
+			 * Transform each `src` into an URI by replacing the `file:./`
```

Comment:
```suggestion
			 * Transforms each 'src' into an URI by replacing 'file:./'
```
- Officially, markdown syntax is only valid in the body of a docblock, not the title.